baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/RedundantModifier.java

```diff
@@ -108,6 +135,12 @@ public Description matchMethod(MethodTree tree, VisitorState state) {
                     .addFix(SuggestedFixes.removeModifiers(tree, state, Modifier.PUBLIC, Modifier.ABSTRACT))
                     .build();
         }
+        if (INTERFACE_STATIC_METHOD_MODIFIERS.matches(tree, state)) {
+            return buildDescription(tree)
+                    .setMessage("Interface components are public by default. The 'public' modifier is unnecessary.")
```

Comment:
interesting that java9 lets people define `private` methods in interfaces! https://bugs.openjdk.java.net/browse/JDK-8071453

This should be all fine though, because we're explicitly only detecting and fixing the `public` keyword.

Reply:
Yep, I designed this with [JEP 213](https://openjdk.java.net/jeps/213) in mind.